include/proteus/Frontend/Func.hpp

```diff
@@ -151,6 +153,13 @@ class FuncBase {
     return Lower(*this);
   }
 
+  ForLoopBuilder forLoop(LoopBoundsDescription Bounds);
+  ForLoopBuilder forLoop(LoopBoundsDescription Bounds,
+                         std::function<void()> Body);
```

Comment:
I'm not fond of the `std::function` and its overhead. I'd suggest to make this a templated Lambda and move to implementation to the header:
```
struct EmptyLambda {
    void operator()() const {}
};

template<typename Lambda>
ForLoopBuilder forLoop(LoopBoundsDescription Bounds,
                         Lambda &&Body = {});
```